### PR TITLE
Vine: --dask-colors and --fontsize options added

### PR DESCRIPTION
**Pull request by JinZhou5042**
## Proposed changes

This PR adds two options into `vine_plot_txn_log.py`, one is `--dask-colors` that enables us to produce a figure in which color palette aligns with Dask's, one is `--fontsize` by which we can set the label size of the figure legend.

## Post-change actions

Put an 'x' in the boxes that describe post-change actions that you have done.
The more 'x' ticked, the faster your changes are accepted by maintainers.

- [x] `make test`       Run local tests prior to pushing.
- [ ] `make format`     Format source code to comply with lint policies. Note that some lint errors can only be resolved manually (e.g., Python)
- [ ] `make lint`       Run lint on source code prior to pushing.
- [ ] Manual Update     Did you update the manual to reflect your changes, if appropriate? This action should be done after your changes are approved but not merged.
- [ ] Type Labels       Select github labels for the type of this change: bug, enhancement, etc.
- [ ] Product Labels    Select github labels for the product affected: TaskVine, Makeflow, etc.
- [ ] PR RTM            Mark your PR as ready to merge.

## Additional comments
The previous PR was accidently closed I messed up the git history, this is a clear one.

### CODE REVIEW

File: taskvine/src/tools/vine_plot_txn_log [hunk 1]
```diff
@@ -565,7 +586,9 @@ class TxnPlot:
     def set_legend(self, fig):
         if self.opts.legend != 'none':
             handles = [mpl.patches.Patch(color=c, label=l) for l,c in self.legend.items()]
-            fig.legend(loc=self.opts.legend, handles=handles)
+
+            fontsize = args.fontsize
```
Comment (btovar) [inline taskvine/src/tools/vine_plot_txn_log:590]: This variable is not needed.

File: taskvine/src/tools/vine_plot_txn_log [hunk 2]
```diff
@@ -687,31 +710,55 @@ class TxnPlotWorkers(TxnPlot):
...
+                else:
+                    base_slot = self.plot_by_worker(fig, current_worker_tasks, current_worker_times, current_worker_transfers, base_slot)
 
         fig.set_ylabel("workers")
         fig.set_yticks((0, base_slot))
-        fig.set_yticklabels(["0", str(gs.ngroups)])  # ytick number of workers
+        fig.set_yticklabels(["0", str(grouped_tasks.ngroups)])  # ytick number of workers
 
-        if not self.opts.expand_waiting:
+        if not args.dask_colors and not self.opts.expand_waiting:
```
Comment (btovar) [inline taskvine/src/tools/vine_plot_txn_log:761]: Since you are using legend_dask directly in plot_dask, you don't need to add dask_colors here.

File: taskvine/src/tools/vine_plot_txn_log [hunk 3]
```diff
@@ -805,6 +851,9 @@ if __name__ == "__main__":
     parser.add_argument('--dpi', nargs='?', type=int, help='output image resoulution', default=300)
     parser.add_argument('--tex', action='store_true', help='use tex fonts')
     parser.add_argument('--nograph', action='store_true', help='do not show a matplotlib window')
+    parser.add_argument('--dask-colors', action='store_true', help='color palette aligns with Dask\'s')
+    parser.add_argument('--fontsize', nargs='?', type=str, help='set fontsize of legend (default medium or 10)', default='medium')
```
Comment (btovar) [inline taskvine/src/tools/vine_plot_txn_log:855]: Remove  "(default medium or 10)". arg parse adds its one default strings,

File: taskvine/src/tools/vine_plot_txn_log [hunk 4]
```diff
@@ -687,31 +710,55 @@ class TxnPlotWorkers(TxnPlot):
...
-
-        base_slot=0
-        for wid in by_min_connects:
-            worker = gsd.get(wid, None)
-            worker_times = ws[ws["worker_id"] == wid]
-            worker_xfers = xs[xs["worker_id"] == wid]
-
-            if worker is not None:
-                base_slot = self.plot_by_worker(fig, worker, worker_times, worker_xfers, base_slot)
+        tasks = manager.tasks
```
Comment (btovar) [inline taskvine/src/tools/vine_plot_txn_log:737]: Submit these changes in another pr.

**Review by btovar — CHANGES_REQUESTED** [review]

File: taskvine/src/tools/vine_plot_txn_log (hunk 4 above)
Comment (JinZhou5042) [reply taskvine/src/tools/vine_plot_txn_log:737]: > Submit these changes in another pr.

I messed the git history again, let me fix this and open a new PR.
